### PR TITLE
Toolchain Updates

### PR DESCRIPTION
**Pull request by antonlacon**
This updates binutils to 2.37 and glibc to 2.34. Patches were refreshed for both. ~~BlueZ gets updated and~~ fuse gets patched for glibc-2.34 compatibility. Gcc and glibc configure options have been updated. Glibc is removing a nonexistent option, and disabling Werror. gcc is removing a nonexistent option and restoring some defaults since GCC5.

Glibc-2.34 will probably be disruptive. From the release announce: https://sourceware.org/pipermail/libc-alpha/2021-August/129718.html

> In order to support smoother in-place-upgrades and to simplify
  the implementation of the runtime all functionality formerly
  implemented in the libraries libpthread, libdl, libutil, libanl has
  been integrated into libc.  New applications do not need to link with
  -lpthread, -ldl, -lutil, -lanl anymore.  For backwards compatibility,
  empty static archives libpthread.a, libdl.a, libutil.a, libanl.a are
  provided, so that the linker options keep working.  Applications which
  have been linked against glibc 2.33 or earlier continue to load the
  corresponding shared objects (which are now empty).  The integration
  of those libraries into libc means that additional symbols become
  available by default.  This can cause applications that contain weak
  references to take unexpected code paths that would only have been
  used in previous glibc versions when e.g. preloading libpthread.so.0,
  potentially exposing application bugs.

Widevine patches have been refreshed for 2.34, but have not been runtime tested.

This also renames GCC_OPTS to TARGET_ARCH_GCC_OPTS to help prevent confusion that this is set in  the config/arch.$ARCH options. I'm open to another name.

Run tested on RPi3.aarch64. Build tested on RPi2.arm. Draft PR so people can test other arches, and I hopefully have time to test moving from glibc's libcrypt to libxcrypt.

### CODE REVIEW

**Comment by heitbaum** [comment]
@antonlacon - I have both binutils/glibc running fine in :dev and on my devices - there are some other fixes in my :dev to support 2.34 - I'll find them for you if not already included.
- https://github.com/LibreELEC/LibreELEC.tv/commit/0766b9403fcc63a94158bf1c3045b20364742924

File: packages/devel/binutils/package.mk [hunk 1]
```diff
@@ -3,8 +3,8 @@
 # Copyright (C) 2018-present Team LibreELEC (https://libreelec.tv)
 
 PKG_NAME="binutils"
-PKG_VERSION="2.35.1"
-PKG_SHA256="3ced91db9bf01182b7e420eab68039f2083aed0a214c0424e257eae3ddee8607"
+PKG_VERSION="2.37"
+PKG_SHA256="820d9724f020a3e69cb337893a0b63c2db161dadcb0e06fc11dc29eb1e84a32c"
 PKG_LICENSE="GPL"
 PKG_SITE="http://www.gnu.org/software/binutils/"
 PKG_URL="http://ftp.gnu.org/gnu/binutils/${PKG_NAME}-${PKG_VERSION}.tar.xz"
```
Comment (heitbaum) [inline packages/devel/binutils/package.mk:10]: Please change PKG_SITE/URL to https

File: packages/devel/glibc/package.mk [hunk 2]
```diff
@@ -3,8 +3,8 @@
 # Copyright (C) 2018-present Team LibreELEC (https://libreelec.tv)
 
 PKG_NAME="glibc"
-PKG_VERSION="2.32"
-PKG_SHA256="1627ea54f5a1a8467032563393e0901077626dc66f37f10ee6363bb722222836"
+PKG_VERSION="2.34"
+PKG_SHA256="44d26a1fe20b8853a48f470ead01e4279e869ac149b195dda4e44a195d981ab2"
 PKG_LICENSE="GPL"
 PKG_SITE="http://www.gnu.org/software/libc/"
 PKG_URL="http://ftp.gnu.org/pub/gnu/glibc/${PKG_NAME}-${PKG_VERSION}.tar.xz"
```
Comment (heitbaum) [inline packages/devel/glibc/package.mk:10]: Please change PKG_SITE/URL to https

**Review by heitbaum — COMMENTED** [review]
Some minors - I checked the rest of my :dev tree and that is all that stood out. There could be some other packages - but not sure.

**Comment by antonlacon** [comment]
What version of Bison was last tested without the glibc patch? 3.8.1 looks like it has a fix for the issue, and I haven't had problems building.

**Comment by heitbaum** [comment]
> What version of Bison was last tested without the glibc patch? 3.8.1 looks like it has a fix for the issue, and I haven't had problems building.

yes - both the "hack fix" and 3.8.1 both work - so lets just go with 3.8.1 and glibc2.34 - hopefully 2.35 does the tidy up to standard for bison - but probably doesnt really matter.

**Comment by heitbaum** [comment]
https://github.com/heitbaum/LibreELEC.tv/commit/6ac18ad1e16717ead2fa98c49aa853e80e0b683c glibc 2.35 compiled up. Though need to rework / review this patch:
- https://github.com/LibreELEC/LibreELEC.tv/blob/master/packages/devel/glibc/patches/glibc-fix-dns-with-broken-routers.patch

ref: https://savannah.gnu.org/forum/forum.php?forum_id=10111

File: packages/devel/glibc/package.mk [hunk 3]
```diff
@@ -19,6 +19,7 @@ PKG_CONFIGURE_OPTS_TARGET="BASH_SHELL=/bin/sh \
                            --libexecdir=/usr/lib/glibc \
                            --cache-file=config.cache \
                            --disable-profile \
+                           --disable-werror \
```
[the remark this answers is not in the document]
Comment (antonlacon) [reply packages/devel/glibc/package.mk:22]: I'm not sure what the benefit is to having glibc error out on a warning for us. My understanding is it's used by glibc developers and glibc shouldn't have errors using the current gcc at the time of its release, and that's why errors show up later sometimes.

Gentoo is the only other distro I know how to quickly look up their build recipe for, and they disable Werror too.

**Comment by antonlacon** [comment]
I've marked it ready. As a reminder, I have only built test the widevine patch - I don't use it. I also run very few addons (usually none), so breakage may be lurking there.

**Review by heitbaum — COMMENTED** [review]
GtG from my testing.

**Comment by knaerzche** [comment]
Runtime tested on RK3399 (also with widevine)

**Comment by SupervisedThinking** [comment]
Good to go?

**Comment by antonlacon** [comment]
No concerns have reached me. Good from my end.

**Comment by SupervisedThinking** [comment]
@antonlacon well looks like grub needs a host build then:
```
config.status: creating grub
make[3]: Verzeichnis „/mnt/dev/LibreELEC-RR/build.LibreELEC-x11.x86_64-11.0-devel/build/grub-635ef55ed1252f92fe3bf70caefd185dcc507c43/util/bash-completion.d“ wird verlassen
make[2]: Verzeichnis „/mnt/dev/LibreELEC-RR/build.LibreELEC-x11.x86_64-11.0-devel/build/grub-635ef55ed1252f92fe3bf70caefd185dcc507c43“ wird verlassen
make[1]: Leaving directory '/mnt/dev/LibreELEC-RR/build.LibreELEC-x11.x86_64-11.0-devel/build/grub-635ef55ed1252f92fe3bf70caefd185dcc507c43'
/mnt/dev/LibreELEC-RR/build.LibreELEC-x11.x86_64-11.0-devel/build/grub-635ef55ed1252f92fe3bf70caefd185dcc507c43/grub-mkimage: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /mnt/dev/LibreELEC-RR/build.LibreELEC-x11.x86_64-11.0-devel/build/grub-635ef55ed1252f92fe3bf70caefd185dcc507c43/grub-mkimage)
/mnt/dev/LibreELEC-RR/build.LibreELEC-x11.x86_64-11.0-devel/build/grub-635ef55ed1252f92fe3bf70caefd185dcc507c43/grub-mkimage: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /mnt/dev/LibreELEC-RR/build.LibreELEC-x11.x86_64-11.0-devel/build/grub-635ef55ed1252f92fe3bf70caefd185dcc507c43/grub-mkimage)
FAILURE: scripts/build grub:target during makeinstall_target (package.mk)
*********** FAILED COMMAND ***********
${PKG_BUILD}/grub-mkimage -d . -o bootia32.efi -O i386-efi -p /EFI/BOOT boot chain configfile ext2 fat linux search efi_gop efi_uga part_gpt gzio gettext loadenv loadbios memrw
**************************************
*********** FAILED COMMAND ***********
${SCRIPTS}/build "${1}" "${PARENT_PKG}"
**************************************
FAILURE: scripts/install grub:target has failed!
```

**Comment by heitbaum** [comment]
> ```
> /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found 
> ```
> 
> 
>      
>   
> 
> @antonlacon well looks like grub needs a host build then:
> 
> 
> ```
> config.status: creating grub
> make[3]: Verzeichnis „/mnt/dev/LibreELEC-RR/build.LibreELEC-x11.x86_64-11.0-devel/build/grub-635ef55ed1252f92fe3bf70caefd185dcc507c43/util/bash-completion.d“ wird verlassen
> make[2]: Verzeichnis „/mnt/dev/LibreELEC-RR/build.LibreELEC-x11.x86_64-11.0-devel/build/grub-635ef55ed1252f92fe3bf70caefd185dcc507c43“ wird verlassen
> make[1]: Leaving directory '/mnt/dev/LibreELEC-RR/build.LibreELEC-x11.x86_64-11.0-devel/build/grub-635ef55ed1252f92fe3bf70caefd185dcc507c43'
> /mnt/dev/LibreELEC-RR/build.LibreELEC-x11.x86_64-11.0-devel/build/grub-635ef55ed1252f92fe3bf70caefd185dcc507c43/grub-mkimage: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /mnt/dev/LibreELEC-RR/build.LibreELEC-x11.x86_64-11.0-devel/build/grub-635ef55ed1252f92fe3bf70caefd185dcc507c43/grub-mkimage)
> /mnt/dev/LibreELEC-RR/build.LibreELEC-x11.x86_64-11.0-devel/build/grub-635ef55ed1252f92fe3bf70caefd185dcc507c43/grub-mkimage: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /mnt/dev/LibreELEC-RR/build.LibreELEC-x11.x86_64-11.0-devel/build/grub-635ef55ed1252f92fe3bf70caefd185dcc507c43/grub-mkimage)
> FAILURE: scripts/build grub:target during makeinstall_target (package.mk)
> *********** FAILED COMMAND ***********
> ${PKG_BUILD}/grub-mkimage -d . -o bootia32.efi -O i386-efi -p /EFI/BOOT boot chain configfile ext2 fat linux search efi_gop efi_uga part_gpt gzio gettext loadenv loadbios memrw
> **************************************
> *********** FAILED COMMAND ***********
> ${SCRIPTS}/build "${1}" "${PARENT_PKG}"
> **************************************
> FAILURE: scripts/install grub:target has failed!
> ```
#5091 fixes it. Only reason I put it back to draft was to look at the reported aarch64 issue. But it is good to merge.

**Comment by antonlacon** [comment]
5091 merged. Were there any other issues or time to merge?

**Comment by heitbaum** [comment]
> 5091 merged. Were there any other issues or time to merge?

All been fine here.

**Review by CvH — APPROVED** [review]

**Pull request merged**